pkg/kube/client.go

```diff
@@ -412,25 +413,61 @@ func updateResource(c *Client, target *resource.Info, currentObj runtime.Object,
 		return nil
 	}
 
-	// if --force is applied, attempt to replace the existing resource with the new object.
-	if force {
+	// update strategies:
+	// default              PATCH
+	// --force              PUT
+	// --recreate           PATCH, on failure DELETE, POST
+	// --recreate --force   DELETE, POST
+	switch {
+	case recreate && force:
+		err = c.deleteAndCreate(helper, target, timeout)
+		if err != nil {
+			return errors.Wrapf(err, "failed to recreate %q with kind %s", target.Name, kind)
+		}
+	case recreate:
+		obj, err = helper.Patch(target.Namespace, target.Name, patchType, patch, nil)
+		if err != nil {
+			if apierrors.IsConflict(err) || apierrors.IsInvalid(err) {
+				err = c.deleteAndCreate(helper, target, timeout)
+			}
+			return errors.Wrapf(err, "failed to recreate %q with kind %s", target.Name, kind)
+		}
+	case force:
 		obj, err = helper.Replace(target.Namespace, target.Name, true, target.Object)
 		if err != nil {
-			return errors.Wrap(err, "failed to replace object")
+			return errors.Wrapf(err, "failed to replace %q with kind $s", target.Name, kind)
```

Comment:
typo: should be `%s` rather than `$s`

Reply:
this has been fixed as:
`return errors.Wrapf(err, "cannot patch %q with kind %s", target.Name, kind)`